modules/swagger-codegen/src/main/resources/ruby/api_client.mustache

```diff
@@ -47,10 +47,17 @@ module {{moduleName}}
       end
 
       unless response.success?
-        fail ApiError.new(:code => response.code,
-                          :response_headers => response.headers,
-                          :response_body => response.body),
-             response.status_message
+        if response.timed_out?
+          fail ApiError.new('Connection timed out')
+        elsif response.code == 0
+          # Errors from libcurl will be made visible here
+          fail ApiError.new(response.return_message)
```

Comment:
Given that the response.code is 0, what about creating the ApiError with `ApiError.new(:code => 0, :message => response.return_message)` instead?